crates/kumod/src/smtp_server.rs

```diff
@@ -1822,6 +1822,15 @@ impl UserData for ConnectionMetaData {
     }
 }
 
+impl<'lua> FromLua<'lua> for ConnectionMetaData {
+    fn from_lua(value: mlua::Value<'lua>, lua: &'lua mlua::Lua) -> mlua::Result<Self> {
+        dbg!(&value);
+        Ok(Self {
+            map: Arc::new(Mutex::new(dbg!(lua.from_value(value)?))),
+        })
+    }
+}
+
```

Comment:
this impl likely conflicts with the regular UserData handling.  I think you should be fine to remove this?

Reply:
It didn't seem to work earlier. Will have another look in a few hours.